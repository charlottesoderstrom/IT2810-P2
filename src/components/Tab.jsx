import React from 'react';

/* Tab is a functional component*/
const Tab = (props) => {
    return ( 
        <div className="tab-container">
            <button id="Gallery1" onClick={()=> props.onTabClick(0)}> Louvre museum</button>
            <button id="Gallery2" onClick= {()=> props.onTabClick(1)}>Munch museum</button>
            <button id="Gallery3" onClick= {()=> props.onTabClick(2)}>Sistine Chapel</button>
            <button id="Gallery4" onClick= {()=> props.onTabClick(3)}>Rijks museum</button>
        </div> );
}
export default Tab;