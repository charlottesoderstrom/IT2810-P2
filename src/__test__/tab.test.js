import React, { Component } from 'react';
import Tab from "../components/Tab";
import renderer from 'react-test-renderer';

/* Jest snapshot test for testing the rendering of the Tab component */
describe('Tab', () => {
  it('renders', () => {
    const component = renderer.create(<Tab />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
