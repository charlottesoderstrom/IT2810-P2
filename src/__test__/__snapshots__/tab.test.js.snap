// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders 1`] = `
<div
  className="tab-container"
>
  <button
    id="Gallery1"
    onClick={[Function]}
  >
     Gallery 1 
  </button>
  <button
    id="Gallery2"
    onClick={[Function]}
  >
    Gallery 2
  </button>
  <button
    id="Gallery3"
    onClick={[Function]}
  >
    Gallery 3
  </button>
  <button
    id="Gallery4"
    onClick={[Function]}
  >
    Gallery 4
  </button>
</div>
`;

exports[`main renders 1`] = `<main />`;
